skiko/src/commonMain/kotlin/org/jetbrains/skiko/SkikoApp.kt

```diff
@@ -0,0 +1,54 @@
+package org.jetbrains.skiko
+
+import org.jetbrains.skia.Canvas
+
+interface SkikoApp {
+    // Input
+    fun onKeyboardEvent(event: SkikoKeyboardEvent)
+    fun onMouseEvent(event: SkikoMouseEvent)
+    fun onInputEvent(event: SkikoInputEvent)
+
+    // Rendering
+    fun onRender(canvas: Canvas, width: Int, height: Int, nanoTime: Long)
+}
+
+open class GenericSkikoApp(
+    val layer: SkiaLayer,
+    val app: SkikoApp): SkikoApp {
+
+    init {
+        layer.setApp(this)
```

Comment:
Probably we should remove this, because it is a side effect during the class creation. The more straightforward way to use this API would be:
```
layer.app = GenericSkikoApp(layer, makeApp())
```